exporter/logicmonitorexporter/logs_exporter.go

```diff
@@ -0,0 +1,127 @@
+)
+
+type logExporter struct {
+	config          *Config
+	logger          *zap.Logger
+	logIngestClient *logs.LMLogIngest
+	settings        component.TelemetrySettings
+}
+
+const (
+	hostname         = "hostname"
+	hostnameProperty = "system.hostname"
+)
+
+// Create new logicmonitor logs exporter
+func newLogsExporter(cfg component.Config, set exporter.CreateSettings) *logExporter {
+	oCfg := cfg.(*Config)
+
+	return &logExporter{
+		config:   oCfg,
+		logger:   set.Logger,
+		settings: set.TelemetrySettings,
+	}
+}
+
+func (e *logExporter) start(_ context.Context, host component.Host) error {
+	client, err := e.config.HTTPClientSettings.ToClient(host, e.settings)
+	if err != nil {
+		return fmt.Errorf("failed to create http client: %w", err)
+	}
+	authParams := utils.AuthParams{
+		AccessID:    e.config.APIToken.AccessID,
+		AccessKey:   e.config.APIToken.AccessKey,
+		BearerToken: e.config.Headers["Authorization"],
+	}
+	options := []logs.Option{
+		logs.WithLogBatchingDisabled(),
+		logs.WithAuthentication(authParams),
+		logs.WithHTTPClient(client),
+		logs.WithEndpoint(e.config.Endpoint),
+	}
+
+	lli, err := logs.NewLMLogIngest(context.Background(), options...)
+	if err != nil {
+		return fmt.Errorf("failed to initialize LMLogIngest: %w", err)
+	}
+	e.logIngestClient = lli
+	return nil
+}
+
+func (e *logExporter) PushLogData(ctx context.Context, lg plog.Logs) error {
+	e.logger.Debug("Pushing log data to Logicmonitor")
+	resourceLogs := lg.ResourceLogs()
+	for i := 0; i < resourceLogs.Len(); i++ {
+		resourceLog := resourceLogs.At(i)
+		libraryLogs := resourceLog.ScopeLogs()
+		for j := 0; j < libraryLogs.Len(); j++ {
+			libraryLog := libraryLogs.At(j)
+			logs := libraryLog.LogRecords()
+			for k := 0; k < logs.Len(); k++ {
+				logMetadataMap := make(map[string]string)
+				resourceMapperMap := make(map[string]string)
+				log := logs.At(k)
+				// Copying resource attributes to log attributes
+				attributesMap := log.Attributes()
+				resourceLog.Resource().Attributes().CopyTo(attributesMap)
+				attributesMap.Range(func(key string, value pcommon.Value) bool {
+					if key == hostname {
+						resourceMapperMap[hostnameProperty] = value.Str()
+					}
+					logMetadataMap[key] = value.Str()
+					return true
+				})
+				e.logger.Debug("Sending log data", zap.String("body", log.Body().Str()), zap.Any("resourcemap", resourceMapperMap), zap.Any("metadatamap", logMetadataMap))
+				payload := translator.ConvertToLMLogInput(log.Body().Str(), timestampFromLogRecord(log).String(), resourceMapperMap, logMetadataMap)
+				err := e.logIngestClient.SendLogs(ctx, payload)
+				if err != nil {
+					e.logger.Error("error while exporting logs ", zap.Error(err), zap.String("body", log.Body().Str()), zap.Any("resourcemap", resourceMapperMap), zap.Any("metadatamap", logMetadataMap))
+					continue
+				}
+				e.logger.Debug("Logs exported successfully!")
+			}
+		}
+	}
+	return nil
+}
+
+func timestampFromLogRecord(lr plog.LogRecord) time.Time {
```

Comment:
See https://pkg.go.dev/go.opentelemetry.io/collector/pdata@v0.66.0/pcommon#Timestamp

Reply:
Sorry I am not able to understand the issue. I tried using log.Timestamp() but it was always returning default values.
Could you please explain what are you trying to suggest?